rust/projects/parallelism/tests/thread_pool.rs

```diff
@@ -0,0 +1,64 @@
+use std::sync::atomic::{AtomicUsize, Ordering};
+use std::sync::Arc;
+
+use kvs::thread_pool::*;
+use kvs::Result;
+
+use crossbeam_utils::sync::WaitGroup;
+
+fn spawn_counter<P: ThreadPool>() -> Result<()> {
+    const TASK_NUM: usize = 20;
+    const ADD_COUNT: usize = 1000;
+
+    let pool = P::new(4)?;
+    let wg = WaitGroup::new();
+    let counter = Arc::new(AtomicUsize::new(0));
+
+    for _ in 0..TASK_NUM {
+        let counter = Arc::clone(&counter);
+        let wg = wg.clone();
+        pool.spawn(move || {
+            for _ in 0..ADD_COUNT {
+                counter.fetch_add(1, Ordering::Relaxed);
+            }
+            drop(wg);
+        })
+    }
+
+    wg.wait();
+    assert_eq!(counter.load(Ordering::SeqCst), TASK_NUM * ADD_COUNT);
+    Ok(())
+}
+
+fn panic_task<P: ThreadPool>() -> Result<()> {
+    const TASK_NUM: usize = 1000;
+
+    let pool = P::new(4)?;
+    for _ in 0..TASK_NUM {
+        pool.spawn(move || {
+            panic!();
+        })
+    }
+
+    spawn_counter::<P>()
+}
```

Comment:
Why is this test running `spawn_counter` at the end? It seems to have no interaction with the panic test.

Reply:
Oh, I wanted to run a counter test in the current pool to check if it still works. I forgot that test actually created a new pool...